index.js

```diff
@@ -1,88 +1,117 @@
 
-module.exports = function() {
-  var plugins = [].slice.call(arguments);
-  var opts = {};
-
-  if (plugins.length > 0 && typeof plugins[0] == "object") {
-    opts = plugins[0];
-    plugins = plugins.slice(1);
+/**
+ * Options
+ *
+ * see: https://github.com/duojs/duo/blob/master/docs/api.md
+ */
+function setOptions(duo, options) {
+  if (!!options.dev || !!options.development) {
+    duo = duo.development(true);
   }
 
-  // TODO should support array form?
-  if (typeof opts.plugins == "object") {
-    var plugs = Object.keys(opts.plugins).map(function(name){
-      var po = opts.plugins[name];
-      return require(name)(po);
-    });
-    plugins = plugs.concat(plugins);
+  if (!!options.sourcemap) {
+    duo = duo.sourcemap(options.sourcemap);
   }
 
-  return function() {
-    return map(function(file, cb) {
-
-      // Start Duo from root
-      // https://github.com/duojs/duo/issues/357
-      var duo = Duo(process.cwd()).entry(file.path);
-
-      setOptions(duo, opts);
+  if (!!options.cache) {
+    duo = duo.cache(true);
+  }
 
-      plugins.forEach(function(plugin) {
-        duo = duo.use(plugin);
-      });
+  if (!!options.copy) {
+    duo = duo.copy(true);
+  }
 
-      return duo.run(function(err, src) {
-          if (err) return cb(err, null);
-          file.contents = new Buffer(src, 'utf8');
-          return cb(null, file);
-      });
-    });
-  };
-};
+  if (!!options.standalone) {
+    duo = duo.standalone(options.standalone);
+  }
 
-// https://github.com/duojs/duo/blob/master/docs/api.md
-function setOptions(duo, opts) {
-  if (!!opts.dev || !!opts.development) {
-    duo = duo.development(true);
+  if (typeof options.concurrency !== 'undefined') {
+    duo = duo.concurrency(options.concurrency);
   }
 
-  if (!!opts.sourcemap) {
-    duo = duo.sourcemap(opts.sourcemap);
+  if (!!options.installTo) {
+    duo = duo.installTo(options.installTo);
   }
 
-  if (!!opts.cache) {
-    duo = duo.cache(true);
+  if (!!options.buildTo) {
+    duo = duo.buildTo(options.buildTo);
   }
 
-  if (!!opts.copy) {
-    duo = duo.copy(true);
+  if (typeof options.token !== 'undefined') {
+    duo = duo.token(options.token);
   }
 
-  // TODO support CSV string value
-  (opts.globals || []).forEach(function(name) {
-    duo = duo.global(name);
-  });
+  return duo;
+}
 
-  if (typeof opts.concurrency !== "undefined") {
-    duo = duo.concurrency(opts.concurrency);
+/**
+ * Export
+ */
+module.exports = function() {
+  /**
+   * Get and assign any arguments
+   */
+  var args = arguments[0];
+  var options = false;
+  var plugins = false;
+
+  if (args && args.plugins) {
+    plugins = args.plugins;
   }
-
-  if (!!opts.installTo) {
-    duo = duo.installTo(opts.installTo);
+  if (args && args.options) {
+    options = args.options;
   }
 
-  // TODO nests assets incorrectly, doesn't update paths correctly
-  if (!!opts.buildTo) {
-    duo = duo.buildTo(opts.buildTo);
+  /**
+   * TODO What's going on here? Needs documentation!
```

Comment:
The code below loads duo plugins, but I think it is better that developer call `require` explicitly, I am sure I've done this by laziness. So I suggest to remove this code at all. But it is better to check that `plugins` is array of functions.